### DISCUSSION TITLE
How do I make the keybinding be detected but not intercepted?

### DISCUSSION THREAD

**Discussion by NightMachinery** [Hammerspoon/hammerspoon#3130] [Q&A] [answered]
I have a keybinding on `b`. This causes the `b` key to not do its original function. 

I have tried doing
```
hs.eventtap.keyStroke({}, 'b')
```
in the keybinding function, but this causes a useless infinite loop.

**Comment by jparoz** [comment, marked as answer]
If you're using the `hs.hotkey` API to make a key binding, it will override any existing behaviour for that key combination. If you just want to monitor key events, [consider using `hs.eventtap.new` instead](http://www.hammerspoon.org/docs/hs.eventtap.html#new). This API allows you to choose whether to post the event or not, by returning either true or false from your callback. You can't directly create an event tap for a single key, but you can use something like the following to approximate the behaviour: 
```lua
bEventTap = hs.eventtap.new({hs.eventtap.event.types.keyDown}, function(event)
    if event:getCharacters() ~= 'b' then return end
    -- do stuff here
    return false  -- don't delete the event
end)
bEventTap:start()
```

**Comment by muescha** [comment]
the other way around is to save the handler of the binding and disable and enable it between the keystroke:

(pseudocode)
```lua
hotkeyHandler = hs.hotkey.bind(modifier, key, message, function()
if something then
   -- do things
 else
   -- do original keystroke
   hotkeyHandler:disable()
   hs.eventtap.keyStroke(modifier, key)
   hotkeyHandler:enable()
 end
```

to avoid the infinite loop you need to disable the `hotkeyHandler`



PS:
I have created some helper with this behaviour: 
### `bindHotkey`
when it should fire only in specific apps or in a specific app and tab (additional functionality: I disabled the logging - otherwise it always shows a log entry if it the handler is disabled/enabled)

Source:
- https://github.com/muescha/dot_hammerspoon/blob/master/Helpers/SendKeysOnlyInApp.lua#L123

Usages:
- https://github.com/muescha/dot_hammerspoon/blob/master/Functions/ChromeTabToNewWindow.lua#L9
- https://github.com/muescha/dot_hammerspoon/blob/master/Functions/ChromeNewWindow.lua#L10
- https://github.com/muescha/dot_hammerspoon/blob/master/Functions/MailMateDisableCmdW.lua#L13

### `HotkeyBindSafe`

which I need to disable my keyhandler with the hyper key so that I can resend a keystroke with fewer modifiers (my hyper is cmd/opt/ctrl/shift but i need to send only  cmd+key):

Source:
- https://github.com/muescha/dot_hammerspoon/blob/master/Helpers/HotkeyBindSafe.lua#L27
Usages:
- https://github.com/muescha/dot_hammerspoon/blob/master/Functions/KeyMapping.lua#L7
